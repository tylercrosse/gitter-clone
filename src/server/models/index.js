import logger from '../config/logger';

const fs = require('fs');
const path = require('path');
const Sequelize = require('sequelize');

const env = process.env.NODE_ENV || 'development';
const config = require('../config/config.json')[env];

const basename = path.basename(module.filename);
const db = {};
let sequelize;

if (config.use_env_variable) {
  sequelize = new Sequelize(process.env[config.use_env_variable]);
} else {
  sequelize = new Sequelize(
    config.database,
    config.username,
    config.password,
    config
  );
}

sequelize
  .authenticate()
  .then(() => {
    logger.log('info', '==> ðŸ›¢ Postgres connected!');
  })
  .catch((err) => {
    logger.error('Unable to connect to the database:', err);
  });

fs.readdirSync(__dirname)
  .filter((file) => {
    return (
      file.indexOf('.') !== 0 && file !== basename && file.slice(-3) === '.js'
    );
  })
  .forEach((file) => {
    const model = sequelize.import(path.join(__dirname, file));
    db[model.name] = model;
  });

Object.keys(db).forEach((modelName) => {
  if (db[modelName].associate) {
    db[modelName].associate(db);
  }
});

db.sequelize = sequelize;
db.Sequelize = Sequelize;

module.exports = db;
