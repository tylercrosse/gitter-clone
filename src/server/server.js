import express        from 'express';
import compression    from 'compression';
import bodyParser     from 'body-parser';
import SocketIo       from 'socket.io';
import expressWinston from 'express-winston';
import logger, {
  errLogger,
  reqLogger }         from './config/logger';
import socketCtlr     from './controllers/sockets';
import router         from './config/routes';

const app = express();
const port = process.env.PORT || 3333;

/* istanbul ignore next: webpack middleware for dev */
if (process.env.NODE_ENV === 'development') {
  const webpackDevMiddlewareInstance = require('./config/devConfig.js').webpackDevMiddlewareInstance;
  const webpackHotMiddlewareInstance = require('./config/devConfig.js').webpackHotMiddlewareInstance;

  app.use(webpackDevMiddlewareInstance);
  app.use(webpackHotMiddlewareInstance);
}

// other middleware
app.use(compression());
app.use(bodyParser.urlencoded({ extended: false }));
app.use(bodyParser.json());
app.use(expressWinston.logger({
  winstonInstance: reqLogger
}));
// serve static files
app.use(express.static('static'));
app.use('/', router);
app.use(expressWinston.errorLogger({
  winstonInstance: errLogger
}));

export const server = app.listen(port, (err) => {
  /* istanbul ignore next: obvious when fails, hard to test */
  if (err) {
    logger.log('error', err);
  }
  let host = server.address().address;
  /* istanbul ignore next: very hard to test */
  if (host === '::') {
    host = '127.0.0.1';
  }
  logger.log('info', '==> ðŸŒŽ Listening at http://%s:%s', host, port);
});

// socket.io config
export const io = new SocketIo(server);
socketCtlr(io);

export default app;
