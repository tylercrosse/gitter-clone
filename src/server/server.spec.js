import request    from 'supertest';
// import mongoose   from 'mongoose';
// import Convo      from './models/Convo';
import { server } from './server';

jest.mock('./config/logger');

describe('express serving', () => {
  afterAll(() => {
    // mongoose.connection.close();
    server.close();
  });

  describe('routing', () => {
    it('should respond to / with the index.html', () => {
      return request(server)
        .get('/')
        .expect(200)
        .expect((res) => expect(res.text).toMatchSnapshot());
    });

    it('should respond to /api/messages with json', () => {
      // const name = 'chat';
      // const options = { upsert: true, new: true, setDefaultsOnInsert: true };
      // return Convo.findOneAndUpdate({ name }, {}, options)
      //   .then(() => {
      //     return request(server)
      //       .get('/api/messages/' + name)
      //       .expect(200)
      //       .expect('Content-Type', /json/);
      //   });
    });

    it('should respond to /api/convos with json', () => {
      // return request(server)
      //   .get('/api/convos')
      //   .expect(200)
      //   .expect('Content-Type', /json/);
    });

    it('should respond to /* with the index.html', () => {
      return request(server)
        .get('/foo/bar')
        .expect(200)
        .expect((res) => expect(res.text).toMatchSnapshot());
    });
  });

  it('should log an error if app.listen fails', () => {

  });
});
