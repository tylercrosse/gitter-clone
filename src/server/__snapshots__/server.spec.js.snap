// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`express serving routing should respond to / with the index.html 1`] = `
"<!doctype html>
<html>
  <head>
    <meta charset=\\"UTF-8\\"/>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0\\" />
    <meta name=\\"theme-color\\" content=\\"#753a88\\">
    <title>Gitter Clone</title>
    <link href=\\"/manifest.json\\" rel=\\"manifest\\">
    <link href=\\"/css/style.min.css\\" rel=\\"stylesheet\\">
    <link href=\\"/font/SourceSansPro-Light.ttf\\" rel=\\"stylesheet\\" type=\\"application/x-font-ttf\\">
    <link href=\\"/font/SourceSansPro-Regular.ttf\\" rel=\\"stylesheet\\" type=\\"application/x-font-ttf\\">
    <link href=\\"/font/SourceSansPro-Semibold.ttf\\" rel=\\"stylesheet\\" type=\\"application/x-font-ttf\\">
  </head>
  <body>
    <div id=\\"root\\"><div class=\\"welcome\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-1359534292\\"><header class=\\"intro-panel\\" data-reactid=\\"2\\"><nav class=\\"site-nav\\" data-reactid=\\"3\\"><div class=\\"wrap\\" data-reactid=\\"4\\"><button class=\\"sign-in-btn\\" data-reactid=\\"5\\">Sign in to start talking</button></div></nav><div class=\\"wrap\\" data-reactid=\\"6\\"><h1 data-reactid=\\"7\\">Welcome to Gitter Clone</h1><p data-reactid=\\"8\\">This is app was built as portfolio piece representing a subset of the features of Troupe Technology&#x27;s wonderful chat app, Gitter.</p><div class=\\"intro-panel-button-block\\" data-reactid=\\"9\\"><a class=\\"intro-panel-button button-caribbean\\" href=\\"/explore\\" data-reactid=\\"10\\">Explore the clone</a><a class=\\"intro-panel-button button-subdued\\" href=\\"https://gitter.im/\\" target=\\"_blank\\" rel=\\"noreferrer noopener\\" data-reactid=\\"11\\">Go to real Gitter.im</a></div><ul class=\\"mini-features-list\\" data-reactid=\\"12\\"><li data-reactid=\\"13\\"><h2 data-reactid=\\"14\\">Built in React</h2><span data-reactid=\\"15\\">Full stack chat app with React powered views.</span></li><li data-reactid=\\"16\\"><h2 data-reactid=\\"17\\">Redux</h2><span data-reactid=\\"18\\">The app state is controlled by Redux.</span></li><li data-reactid=\\"19\\"><h2 data-reactid=\\"20\\">Markdown Support</h2><span data-reactid=\\"21\\">Messages support bursts and git flavoured markdown.</span></li></ul></div></header><section class=\\"tech-panel\\" data-reactid=\\"22\\"><div class=\\"wrap\\" data-reactid=\\"23\\"><h1 data-reactid=\\"24\\">Built with</h1><ul class=\\"tech-icons\\" data-reactid=\\"25\\"><li data-reactid=\\"26\\"><img class=\\"react-logo\\" src=\\"./img/react.svg\\" alt=\\"react logo\\" data-reactid=\\"27\\"/><figcaption data-reactid=\\"28\\">React</figcaption></li><li data-reactid=\\"29\\"><img class=\\"redux-logo\\" src=\\"./img/redux.svg\\" alt=\\"redux logo\\" data-reactid=\\"30\\"/><figcaption data-reactid=\\"31\\">Redux</figcaption></li><li data-reactid=\\"32\\"><img class=\\"webpack-logo\\" src=\\"./img/webpack.svg\\" alt=\\"webpack logo\\" data-reactid=\\"33\\"/><figcaption data-reactid=\\"34\\">Webpack</figcaption></li><li data-reactid=\\"35\\"><img class=\\"express-logo\\" src=\\"./img/express.svg\\" alt=\\"express logo\\" data-reactid=\\"36\\"/><figcaption data-reactid=\\"37\\">Express</figcaption></li><li data-reactid=\\"38\\"><img class=\\"mongodb-logo\\" src=\\"./img/mongodb.svg\\" alt=\\"mongodb logo\\" data-reactid=\\"39\\"/><figcaption data-reactid=\\"40\\">MongoDB</figcaption></li><li data-reactid=\\"41\\"><img class=\\"jest-logo\\" src=\\"./img/jest.svg\\" alt=\\"jest logo\\" data-reactid=\\"42\\"/><figcaption data-reactid=\\"43\\">Jest</figcaption></li><li data-reactid=\\"44\\"><img class=\\"semaphor-logo\\" src=\\"./img/semaphor.svg\\" alt=\\"semaphor logo\\" data-reactid=\\"45\\"/><figcaption data-reactid=\\"46\\">SemaphoreCI</figcaption></li><li data-reactid=\\"47\\"><img class=\\"digitalocean-logo\\" src=\\"./img/digitalocean.svg\\" alt=\\"digitalocean logo\\" data-reactid=\\"48\\"/><figcaption data-reactid=\\"49\\">Digital Ocean</figcaption></li></ul></div></section><section class=\\"endoresment-panel\\" data-reactid=\\"50\\"><div class=\\"wrap\\" data-reactid=\\"51\\"><h1 data-reactid=\\"52\\">Check out the real Gitter.im</h1><p data-reactid=\\"53\\"><!-- react-text: 54 -->This is just a fan application. For the real Gitter messaging applcation go to<!-- /react-text --><a href=\\"https://gitter.im/\\" target=\\"_blank\\" rel=\\"noreferrer noopener\\" data-reactid=\\"55\\"> gitter.im</a></p></div></section><footer class=\\"bottom-panel\\" data-reactid=\\"56\\"><!-- react-text: 57 -->Built with <!-- /react-text --><span class=\\"heart\\" data-reactid=\\"58\\">&lt;3</span><!-- react-text: 59 --> by<!-- /react-text --><a href=\\"http://tylercrosse.com/\\" target=\\"_blank\\" rel=\\"noreferrer noopener\\" data-reactid=\\"60\\">Tyler Crosse</a><div class=\\"links\\" data-reactid=\\"61\\"><a href=\\"https://github.com/tylercrosse/gitter-clone\\" target=\\"_blank\\" rel=\\"noreferrer noopener\\" data-reactid=\\"62\\">Github</a><a href=\\"https://www.linkedin.com/in/tylercrosse\\" target=\\"_blank\\" rel=\\"noreferrer noopener\\" data-reactid=\\"63\\">LinkedIn</a></div></footer><noscript data-reactid=\\"64\\"></noscript></div></div>
    <script>
      window.__PRELOADED_STATE__ = {\\"messages\\":{},\\"convos\\":{},\\"user\\":{\\"loggedIn\\":false},\\"ui\\":{\\"modalIsOpen\\":{\\"createRoom\\":false,\\"signIn\\":false},\\"isFetching\\":false,\\"usersTyping\\":[]},\\"routing\\":{\\"locationBeforeTransitions\\":null},\\"error\\":{\\"messages\\":false,\\"convos\\":false,\\"signIn\\":false}}
    </script>
    <script src=\\"/js/manifest.js\\"></script>
    <script src=\\"/js/vendor.js\\"></script>
    <script src=\\"/js/app.js\\"></script>
    <script type=\\"text/javascript\\">
      if ('serviceWorker' in navigator) {
        window.addEventListener('load', function() {
          navigator.serviceWorker.register('/js/serviceWorker.js');
        });
      }
    </script>
  </body>
</html>"
`;

exports[`express serving routing should respond to /* with the index.html 1`] = `
"<!doctype html>
<html>
  <head>
    <meta charset=\\"UTF-8\\"/>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0\\" />
    <meta name=\\"theme-color\\" content=\\"#753a88\\">
    <title>Gitter Clone</title>
    <link href=\\"/manifest.json\\" rel=\\"manifest\\">
    <link href=\\"/css/style.min.css\\" rel=\\"stylesheet\\">
    <link href=\\"/font/SourceSansPro-Light.ttf\\" rel=\\"stylesheet\\" type=\\"application/x-font-ttf\\">
    <link href=\\"/font/SourceSansPro-Regular.ttf\\" rel=\\"stylesheet\\" type=\\"application/x-font-ttf\\">
    <link href=\\"/font/SourceSansPro-Semibold.ttf\\" rel=\\"stylesheet\\" type=\\"application/x-font-ttf\\">
  </head>
  <body>
    <div id=\\"root\\"><div class=\\"generic404\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-265921900\\"><h1 data-reactid=\\"2\\">404</h1><p data-reactid=\\"3\\">This is not the chat you are looking for,</p><a href=\\"/explore\\" data-reactid=\\"4\\">try exploring existing chats.</a><p data-reactid=\\"5\\">Or create a new chat</p></div></div>
    <script>
      window.__PRELOADED_STATE__ = {\\"messages\\":{},\\"convos\\":{},\\"user\\":{\\"loggedIn\\":false},\\"ui\\":{\\"modalIsOpen\\":{\\"createRoom\\":false,\\"signIn\\":false},\\"isFetching\\":false,\\"usersTyping\\":[]},\\"routing\\":{\\"locationBeforeTransitions\\":null},\\"error\\":{\\"messages\\":false,\\"convos\\":false,\\"signIn\\":false}}
    </script>
    <script src=\\"/js/manifest.js\\"></script>
    <script src=\\"/js/vendor.js\\"></script>
    <script src=\\"/js/app.js\\"></script>
    <script type=\\"text/javascript\\">
      if ('serviceWorker' in navigator) {
        window.addEventListener('load', function() {
          navigator.serviceWorker.register('/js/serviceWorker.js');
        });
      }
    </script>
  </body>
</html>"
`;
